src/ble/att/mod.rs

```diff
@@ -268,14 +273,14 @@ impl Default for AttPermission {
 
 /// Trait for attribute sets that can be hosted by an `AttributeServer`.
 pub trait Attributes {
-    fn attributes(&self) -> &[Attribute];
+    fn attributes(&mut self) -> &[Attribute];
```

Comment:
I know we've talked about this already, but if this trait is modified to look like this instead, perhaps we can avoid some problems wrt storing the attributes in the future:

```rust
trait Attributes<'a> {
    type Iter: Iterator<Item=&'a mut Attribute>;
    fn attributes(&'a mut self) -> Self::Iter;
}
```

This can also be done later though, just wanted to write down what I had in mind here.